src/pages/deti.tsx

```diff
@@ -135,18 +127,116 @@ const Deti: NextPage<Props> = ({
   )
 }
 
-// const validationSchema = Yup.object().shape<ChildrenUserInput>({
-//   children: Yup.boolean()
-//     .required()
-//     .nullable(),
-//   r034: Yup.mixed().when('childs', {
-//     is: true,
-//     then: Yup.mixed(),
-//   }),
-//   // R039: Yup.number().when("employed", {
-//   //   is: true,
-//   //   then: Yup.number().required(),
-//   // }),
-// });
+interface ChildFormProps {
+  index: number
+  savedValues: ChildInput
+}
+const ChildForm = ({ savedValues, index }: ChildFormProps) => {
+  return (
+    <>
+      <Input
+        name={`children[${index}].priezviskoMeno` as any}
+        type="text"
+        label="Meno a priezvisko"
+        width="auto"
+      />
+      <Input
+        name={`children[${index}].rodneCislo` as any}
+        type="text"
+        label="Rodné číslo"
+        width="auto"
+      />
+      <div className="govuk-form-group">
+        <legend className="govuk-fieldset__legend govuk-fieldset__legend--s">
+          <h1 className="govuk-fieldset__heading">
+            Daňový bonus podľa § 33 zákona uplatňujem v mesiacoch
+          </h1>
+        </legend>
+        <div className="govuk-checkboxes">
+          <CheckboxSmall
+            name={`children[${index}].wholeYear`}
+            label="Za celý kalendárny rok"
+          />
+        </div>
+      </div>
+      <div
+        className={classnames('govuk-form-group', styles.inlineFieldContainer)}
+      >
+        <Select
+          name={`children[${index}].monthFrom`}
+          label="Od"
+          className={styles.inlineField}
+          options={monthNames}
+          disabled={savedValues.wholeYear ? 0 : false}
+        />
+        <Select
+          name={`children[${index}].monthTo`}
+          label="Do"
+          className={styles.inlineField}
+          options={monthNames}
+          disabled={savedValues.wholeYear ? 11 : false}
+        />
+      </div>
+    </>
+  )
+}
+
+interface ChildFormErrors {
```

Comment:
Toto by mohlo byt aj `Pick<ChildrenUserInput, 'priezviskoMeno', 'rodneCislo', 'monthTo'>` ale moze byt aj takto

Reply:
Takto:

```
type ChildFormErrors = Pick<
  ChildInput,
  'priezviskoMeno' | 'rodneCislo' | 'monthTo'
>
```

Ale `ChildInput` nema optional props, takze tu je potom type error:
```
const childErrors: ChildFormErrors = {}

# TS2739: Type '{}' is missing the following properties from type 'Pick<ChildInput, "priezviskoMeno" | "rodneCislo" | "monthTo">': priezviskoMeno, rodneCislo, monthTo
```

Reply:
Tak potom
```
const childErrors: Partial<ChildFormErrors> = {}

```
Ale tak netreba to zbytocne komplikovat